crates/sui-graphql-rpc/src/types/transaction_block.rs

```diff
@@ -297,102 +305,212 @@ impl TransactionBlock {
+            .execute_repeatable(move |conn| {
+                // The min or first `tx_sequence_number` of a checkpoint is the previous
+                // checkpoint's `network_total_transactions`. Because this refers to a historical
+                // checkpoint, if we yield a `None` result, we can return early.
+                let lo_tx = match lo_cp {
+                    0 => 0,
+                    _ => {
+                        let sequence_number: Option<i64> = conn
+                            .first(move || {
+                                cp::checkpoints
+                                    .select(cp::network_total_transactions)
+                                    .filter(cp::sequence_number.eq((lo_cp - 1) as i64))
+                            })
+                            .optional()?;
+
+                        match sequence_number {
+                            Some(sequence_number) => sequence_number as u64,
+                            None => return Ok::<_, diesel::result::Error>((false, false, vec![])),
                         }
-
-                        query
-                    },
-                )
+                    }
+                };
+
+                // The max or last `tx_sequence_number` of a checkpoint is the current checkpoint's
+                // `network_total_transactions` - 1. Since we cap the upper bound to
+                // `checkpoint_viewed_at`, we should always yield a result; thus, we can also return
+                // early if result is `None`.
+                let hi_tx = {
+                    let sequence_number: Option<i64> = conn
+                        .first(move || {
+                            cp::checkpoints
+                                .select(cp::network_total_transactions)
+                                .filter(cp::sequence_number.eq(hi_cp as i64))
+                        })
+                        .optional()?;
+
+                    match sequence_number {
+                        Some(sequence_number) => (sequence_number as u64).saturating_sub(1),
+                        None => return Ok::<_, diesel::result::Error>((false, false, vec![])),
+                    }
+                };
+
+                let sender = filter.sign_address;
+
+                let mut bound = TxLookupBound::from_range((Some(lo_tx), Some(hi_tx)));
+
+                // If `transaction_ids` is specified, we can use that in lieu of the range under the
+                // assumption that it will further constrain the rows we look up.
+                if let Some(txs) = &filter.transaction_ids {
+                    let transaction_ids: Vec<TxLookup> =
+                        conn.results(move || select_ids(txs, &bound).into_boxed())?;
+                    if transaction_ids.is_empty() {
+                        return Ok::<_, diesel::result::Error>((false, false, vec![]));
+                    }
+                    bound = TxLookupBound::from_set(
```

Comment:
What happens if both transaction ids and a range (scan limit) is provided? Here we would just be ignoring the range.

Reply:
It's applied when we query transactions with the list of transaction ids